files/grest/rpc/stake_address/rewards_history.sql

```diff
@@ -0,0 +1,41 @@
+CREATE OR REPLACE FUNCTION grest.rewards_history (_stake_address_bech32 text DEFAULT NULL, _epoch_no numeric DEFAULT NULL)
+    RETURNS json STABLE
+    LANGUAGE PLPGSQL
+    AS $$
+BEGIN
+    IF _epoch_no IS NULL THEN
+        RETURN (
+            SELECT
+                json_agg(js) json_final
+            FROM (
+                SELECT
+                    json_build_object('epoch_no', reward.epoch_no, 'stake_pool', pool_hash.view, 'reward_amount', ROUND((reward.amount / 1000000), 2)) js
+                FROM
+                    reward
+                    INNER JOIN stake_address ON reward.addr_id = stake_address.id
+                    INNER JOIN pool_hash ON reward.pool_id = pool_hash.id
+                WHERE
+                    stake_address.view = _stake_address_bech32
+                ORDER BY
+                    epoch_no ASC) t);
+    ELSE
+        RETURN (
+            SELECT
+                json_agg(js) json_final
+            FROM (
+                SELECT
+                    json_build_object('epoch_no', reward.epoch_no, 'stake_pool', pool_hash.view, 'reward_amount', ROUND((reward.amount / 1000000), 2)) js
+                FROM
+                    reward
+                    INNER JOIN stake_address ON reward.addr_id = stake_address.id
+                    INNER JOIN pool_hash ON reward.pool_id = pool_hash.id
+                WHERE
+                    stake_address.view = _stake_address_bech32
```

Comment:
Just noticed filter is missing  , tho epoch_no is now divided into earned/spendable

Reply:
Yep, got it working last night, was just waiting for the ada/lovelace decision. Will update soon

Reply:
ye - avoiding ADA keeps it consistent with chain unit too

Reply:
Updated now, along with adding a type for the rewards according to: 
https://github.com/input-output-hk/cardano-db-sync/commit/5965ee3e87730ec3f8c93b9287320c11f94d2160

> `member` or `leader` should have  `poolId` entry while
> for `treasury` or `reserves` should always have NULL for the `poolId` field.

Some things to clear up:
1. If `addr...` is passed instead of `stake...`, do we want to do a lookup or require only stake address to be passed in?
2. If an epoch is specified, do we consider it as the `spendable` or `earned` epoch for filtering (currently `earned`)?

Reply:
1. For now, my view is prefer to only have stake... address - keeping it simple - until dbsync stabilises in terms of changes, as addr to stake addr mapping is also currently available seperately
2. earned indeed

Reply:
In that case, I think this is ready for removing the draft label